Sources/StreamChat/Database/DTOs/UserDTO.swift

```diff
@@ -21,13 +21,35 @@ class UserDTO: NSManagedObject {
     
     @NSManaged var flaggedBy: CurrentUserDTO?
     
+    @NSManaged var payloadHash: Int64
+    
     /// Returns a fetch request for the dto with the provided `userId`.
     static func user(withID userId: UserId) -> NSFetchRequest<UserDTO> {
         let request = NSFetchRequest<UserDTO>(entityName: UserDTO.entityName)
         request.sortDescriptors = [NSSortDescriptor(keyPath: \UserDTO.id, ascending: false)]
         request.predicate = NSPredicate(format: "id == %@", userId)
         return request
     }
+    
+    override func willSave() {
+        super.willSave()
+        var hasher = Hasher()
+        hasher.combine(id)
+        hasher.combine(name)
+        hasher.combine(imageURL)
+        hasher.combine(userRoleRaw)
+        hasher.combine(userCreatedAt)
+        hasher.combine(userUpdatedAt)
+        hasher.combine(lastActivityAt)
+        hasher.combine(isOnline)
+        // hasher.combine(isInvisible)
+        hasher.combine(isBanned)
+        // hasher.combine(teams)
+        hasher.combine(extraData)
+        // Direct assignment cannot be used
+        // Since it'll generate new `willSave` calls causing recursion
+        assignIfDifferent(self, \.payloadHash, Int64(hasher.finalize()))
+    }
```

Comment:
... or even better - what if we created something like `struct UserHasher: Hashable { ... }` and use it just to get the hash from that? When a new field is added, it's initializer will help us to add it everywhere where needed.

Reply:
actually, great idea